lib/ui/dart_runtime_hooks.cc

```diff
@@ -199,19 +200,28 @@ void Logger_PrintString(Dart_NativeArguments args) {
     // Write to the logcat on Android.
     __android_log_print(ANDROID_LOG_INFO, logger_prefix.c_str(), "%.*s",
                         (int)length, chars);
-#elif defined(OS_IOS)
-    // Write to syslog on iOS.
-    //
+#elif defined(__APPLE__)
     // TODO(cbracken): replace with dedicated communication channel and bypass
     // iOS logging APIs altogether.
-    syslog(1 /* LOG_ALERT */, "%.*s", (int)length, chars);
+    //
+    // Unified logging (os_log) became available in iOS 9.0 and syslog stopped working in iOS 13.0.
+    // idevicesyslog made device syslog available on the connected host, but there is
+    // no known API to view device unified logging on the host.
+    // Flutter tool will continue to observe syslog on devices older than iOS 13.0 since it
+    // provides more logging context, particularly for application crashes.
+    if (__builtin_available(iOS 13.0, macOS 10.11, *)) {
+      static os_log_t dart_log = os_log_create("io.flutter", "dart");
+      os_log(dart_log, "%.*s", (int)length, chars);
```

Comment:
Please avoid C style casts in C++ TU's (I realize the old code ignored this guidance).